share/spack/gitlab/cloud_pipelines/.gitlab-ci.yml

```diff
@@ -253,6 +253,41 @@ build_systems-develop-build:
     - artifacts: True
       job: build_systems-develop-generate
 
+#########################################
+# Development tools stack
+#########################################
+.dev_tools:
+  variables:
+    SPACK_CI_STACK_NAME: dev_tools
```

Comment:
Because of how this is used [here](https://github.com/spack/spack/blob/b28b24ccf8d2900b63b6c560e2a55fd2e17d2f66/share/spack/gitlab/cloud_pipelines/.gitlab-ci.yml#L29), the stack name needs to match the directory in which the stack lives.  So this either needs to be `dev-tools`, or the stack should live in the `dev_tools` subdirectory.